shell/public/offer-template.html

```diff
@@ -10,20 +10,92 @@
   border-spacing: 0;
   border: 0;
 }
-pre {
+#text {
   margin: 0;
   padding: 0;
 }
+#clipboard {
+  display: none;
+  position: absolute;
+  width: 16px;
+  height: 16px;
+  background: #FFFFFF url(clippy.svg) 50% 50% no-repeat;
+  cursor: pointer;
+  border: 0;
+}
+#clipboard:focus {
+  outline: 0;
+}
+#clipboardSuccess {
+  display: none;
+  position: absolute;
+  top: 0;
+  font-family: monospace;
+  background-color: #000;
+  color: #FFF;
+  padding: 1px 4px;
+  border-radius: 2px;
+}
+
+.clipboardLeft #clipboard {
+  display: block;
+  top: 0;
+  left: 0;
+}
+.clipboardLeft #container {
+  overflow: auto;
+  margin-left: 20px;
+}
+.clipboardLeft #clipboardSuccess {
+  left: 20px;
+}
+.clipboardRight #clipboard {
+  display: block;
+  top: 0;
+  right: 0;
+}
+.clipboardRight #container {
+  overflow: auto;
+  margin-right: 20px;
+}
+.clipboardRight #clipboardSuccess {
+  right: 20px;
+}
     </style>
   </head>
   <body>
-    <pre id="text"></pre>
+    <div id="container">
+      <pre id="text"></pre>
+    </div>
+    <button id="clipboard" label="Copy" data-clipboard-target="#text"></button>
```

Comment:
I could be mistaken, but I think you want `aria-label` here, not `label`.  I couldn't find an attribute `label` anywhere in the HTML5 spec.